main.go

```diff
@@ -155,109 +135,64 @@ func nsmClientFactory(ctx context.Context, rootConf *config.Config) func(...netw
 			))...,
 	)
 	if err != nil {
-		log.FromContext(ctx).Fatalf("failed to dial NSM: %v", err.Error())
+		logger.Fatalf("failed dial to NSMgr: %v", err.Error())
 	}
-
 	// ********************************************************************************
 	// Create Network Service Manager nsmClient
 	// ********************************************************************************
-	tokenClient := token.NewClient()
-	sendfdClient := sendfd.NewClient()
+	nsmClient := client.NewClient(ctx,
+		cc,
+		client.WithName(rootConf.Name),
+		client.WithAuthorizeClient(authorize.NewClient()),
+		client.WithAdditionalFunctionality(
+			mechanisms.NewClient(map[string]networkservice.NetworkServiceClient{
+				vfiomech.MECHANISM:   chain.NewNetworkServiceClient(vfio.NewClient(), sriovtoken.NewClient()),
```

Comment:
It seems sriovtoken.NewClient() it is required for kernelmech.MECHANISM scenario as well.